ecosystem/sep-0038.md

```diff
@@ -0,0 +1,200 @@
+`price` | string | The price offered by the anchor for one unit of `asset` in terms of `sell_asset`.
+`decimals` | integer | The number of decimals needed to represent `asset`.
+
+### GET Price
+
+This endpoint can be used to fetch the indicative price for a given asset pair.
+
+#### Request
+
+Name | Type | Description
+-----|------|------------
+`sell_asset` | string | The asset the client would like to sell. Ex. `USDC:circle.io`, `iso4217:ARS`
+`buy_asset` | string | The asset the client would like to exchange `sell_asset` for.
+`sell_amount` | string | The amount of `sell_asset` the client would like to sell for `buy_asset`.
+
+#### Response
+
+Name | Type | Description
+-----|------|------------
+`price` | string | The price offered by the anchor for one unit of `buy_asset` in terms of `sell_asset`.
+
+### POST Quote
+
+This endpoint can be used to request a [firm](https://www.investopedia.com/terms/f/firmquote.asp) quote for a Stellar asset and off-chain asset pair. 
+
+In contrast with the `GET price(s)` endpoints, the amount requested must be held in reserve and not used in calculations of subsequent quotes until the expiration provided in the response.
+
+#### Protecting Against Bad Actors
+
+To protect against bad actor clients reserving all available captial without following through with the requested trades, servers should only accept requests for quotes from Stellar accounts that belong to entities or individuals that have been properly KYC'ed, either via [SEP-12](sep-0012.md) or some other mechanism.
+
+Servers may deny access to the API if misuse is detected.
+
+#### Transaction Fees
+
+It should be expected that the provided quote includes a margin for the provider as a service fee, and this margin will vary depending on the directional flow of funds.
+
+If the client requests some amount of an off-chain asset for providing some amount of a Stellar asset, the client will submit a Stellar transaction delievering funds to the anchor before the expiration included in the response, paying a fee as determined by state of the Stellar network. 
+
+In the reverse scenario, the anchor will submit a Stellar transaction to deliver funds to the client as long as the client delivered off-chain funds to the anchor before the expiration. In this case, the anchor will likely increase their margin to cover the cost of submitting the transaction.
+
+#### Request
+
+Name | Type | Description
+-----|------|------------
+`sell_asset` | string | Same as the definition of `sell_asset` in `GET /price`.
+`buy_asset` | string | Same as the definition of `buy_asset` in `GET /price`.
+`sell_amount` | string | Same as the definition of `sell_amount` in `GET /price`.
+`expire_after` | [UTC ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) string | (optional) The earliest date and time the client can deliver funds to the anchor. Anchors should return `400 Bad Request` if the requested expiration cannot be provided.
```

Comment:
🤔 I find expire_after confusing. I mentioned this [here](https://github.com/stellar/stellar-protocol/pull/908#discussion_r613611275) but I think I did a poor job of explaining.

When I use money transfer services to transfer money between countries it is really common for that service to provide two methods:
1. Immediate transfer where the price I'm given starts now and expires after X hours.
2. A scheduled transfer where the price is unknown. An example of a product that supports this is Wise, they discuss how it works [here](https://wise.com/help/articles/2978063/what-are-scheduled-transfers). One thing they say in their FAQ is:
   >...you won’t get a guaranteed rate. This is because you’re requesting that we send the money out later, and we can’t predict the future — as much as we wish we could.

The `expires_after` field reads like a scheduled transfer. I'm not convinced a SEP-38 implementer will be able to predict the future and provide a guaranteed rate in the future. 

I think `expires_after` was added in response to @erasmus's feedback [here](https://github.com/stellar/stellar-protocol/issues/901#issuecomment-819495250) but I think we may have misunderstood their request:
> I was thinking that the quote requester should be able to specify a desired expiration and get a quote from provider based on that time horizon. As a requester I may want to pay more for a longer expiry time.

I think @erasmus is requesting that the client can specify how long till the rate expires from now as a way to signal to the server they're willing to pay a premium to get longer to pay.

@erasmus please correct me if I'm incorrect.

Reply:
> I'm not convinced a SEP-38 implementer will be able to predict the future and provide a guaranteed rate in the future.

I'm a bit confused by this statement, because the price an anchor provides is guaranteed until `expires_at` as the protocol is written now. It is very possible that the market price for the asset pair changes before `expires_at`, such that the market price would be better for the anchor that the price quoted in the API response, but thats the name of the game. 

Anchors will likely provide more conservative rates for more volatile assets, or in cases where the `delivery_method` specified by the client is slower, allowing for more time for the market price to change.

> I think @erasmus is requesting that the client can specify how long till the rate expires from now as a way to signal to the server they're willing to pay a premium to get a rate that will be guaranteed for longer.

That was my intention with `expire_after`; the client using this parameter would be asking the anchor to provide an `expires_at` value on or after `expires_after`. The rate is still provided immediately and only valid before `expires_at`.

We could be more explicit by renaming the attribute to something like `desired_expiration`?

Reply:
The name `expire_after` is great, it sounds like it does what I was asking, and not what I was thinking it did.

I was confused by the description and your comment [here](https://github.com/stellar/stellar-protocol/pull/908#discussion_r613619017).

❗ I think the description needs changing. It is not the earliest the client can provide the funds, but the latest the clients wishes to be able to provide the funds. The clients wants up to that amount of time at least to provide the funds.

Reply:
Ah I see, thanks for poking at that again. I think the term 'latest' is confusing because the anchor could provide an `expires_at` value that is later than the requested `expire_after` value. How about this description?

> The client's desired `expires_at` date and time for the quote. Anchors should return `400 Bad Request` if the an expiration on or after the requested value cannot be provided.

Reply:
I love it that description.